### PR TITLE
Move CreateTransaction to serializable worker message/task

### PR DESCRIPTION
**Pull request by deekerno**
## Summary
This PR moves the CreateTransaction task into the new serialization format.

## Testing Plan
Added test, and ran the node to ensure proper function.

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ironfish/src/workerPool/tasks/createTransaction.test.slow.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  })
+  const nodeTest = createNodeTest()
+
+  describe('execute', () => {
+    it('creates the transaction', async () => {
+      const account = await useAccountFixture(nodeTest.accounts)
+      const minerTransaction = await useMinersTxFixture(nodeTest.accounts, account)
+
+      const spendNote = minerTransaction.getNote(0).decryptNoteForOwner(account.incomingViewKey)
+      if (spendNote === undefined) {
```
Comment (rohanjadvani) [inline ironfish/src/workerPool/tasks/createTransaction.test.slow.ts:109]:
```suggestion
Assert.isNotUndefined(spendNote)
```

File: ironfish/src/workerPool/tasks/createTransaction.test.slow.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        throw new Error('spendNote should not be undefined')
+      }
+      for (let i = 0; i < minerTransaction.notesLength(); i++) {
+        const note = minerTransaction.getNote(i)
+        await tree.add(note)
+      }
+
+      const authPath = (await tree.witness(0))?.authenticationPath
+      if (authPath === undefined) {
```
Comment (rohanjadvani) [inline ironfish/src/workerPool/tasks/createTransaction.test.slow.ts:118]:
```suggestion
      Assert.isNotUndefined(authPath)
```

File: ironfish/src/workerPool/tasks/createTransaction.test.slow.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            authPath,
+          },
+        ],
+        [{ publicAddress: account.publicAddress, amount: BigInt(1), memo: '' }],
+      )
+
+      const response = task.execute(request)
+      expect(response).toEqual(
+        new CreateTransactionResponse(
+          Buffer.from(response.serializedTransactionPosted),
```
Comment (rohanjadvani) [inline ironfish/src/workerPool/tasks/createTransaction.test.slow.ts:141]: We should make the assertion more strict. Right now the test is reconstructing a response object from a field the response already has, so the test is covering the object itself versus the actual execution logic. 

To have a more explicit assertion, we can (1) mock what `transaction.post` returns and assert that is in the response or (2) reconstruct the serialized transaction and verify that's what is returned. (1) is probably easier, and we can also verify `spend` and `receive` were called. 

There is an example for `createMinersFee` [here](https://github.com/iron-fish/ironfish/blob/serialize-worker-pool/ironfish/src/workerPool/tasks/createMinersFee.test.ts#L57-L60).

File: ironfish/src/workerPool/tasks/createTransaction.ts [hunk 4]
```diff
@@ -24,39 +28,210 @@ export type CreateTransactionRequest = {
...
+
+  serialize(): Buffer {
+    const bw = bufio.write(this.getSize())
+    bw.writeVarString(this.spendKey)
+    bw.writeVarBytes(BigIntUtils.toBytesBE(this.transactionFee))
+    bw.writeU64(this.expirationSequence)
+    bw.writeU64(this.spends.length)
+
+    for (const spend of this.spends) {
+      bw.writeVarBytes(Buffer.from(spend.note))
```
Comment (rohanjadvani) [inline ironfish/src/workerPool/tasks/createTransaction.ts:62]: Does this work?

```suggestion
      bw.writeVarBytes(spend.note)
```

File: ironfish/src/workerPool/tasks/createTransaction.ts [hunk 5]
```diff
@@ -24,39 +28,210 @@ export type CreateTransactionRequest = {
...
+
+      for (const step of spend.authPath) {
+        switch (step.side) {
+          case Side.Left:
+            bw.writeU8(0)
+            break
+          case Side.Right:
+            bw.writeU8(1)
+            break
+          default:
```
Comment (rohanjadvani) [inline ironfish/src/workerPool/tasks/createTransaction.ts:75]: Do we need this `default` branch?

**Review by rohanjadvani — CHANGES_REQUESTED** [review]

File: ironfish/src/workerPool/tasks/createTransaction.ts (hunk 4 above)
Comment (deekerno) [reply ironfish/src/workerPool/tasks/createTransaction.ts:62]: It seems that the tests in `accounts.test.slow.ts` fail if `spend.note` is not explicitly a `Buffer`.

File: ironfish/src/workerPool/tasks/createTransaction.ts (hunk 5 above)
Comment (deekerno) [reply ironfish/src/workerPool/tasks/createTransaction.ts:75]: I placed it there to make sure to mirror some of the other switch-case uses in the code, but removing it has no ill effects.

**Review by rohanjadvani — APPROVED** [review]

**Pull request merged**